gcloud-java-compute/src/main/java/com/google/cloud/compute/RegionId.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.compute;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects.ToStringHelper;
+
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A Google Compute Engine region identity.
+ */
+public final class RegionId extends ResourceId {
+
+  static final Function<String, RegionId> FROM_URL_FUNCTION = new Function<String, RegionId>() {
+    @Override
+    public RegionId apply(String pb) {
+      return RegionId.fromUrl(pb);
+    }
+  };
+  static final Function<RegionId, String> TO_URL_FUNCTION = new Function<RegionId, String>() {
+    @Override
+    public String apply(RegionId regionId) {
+      return regionId.selfLink();
+    }
+  };
+
+  private static final String REGEX = ResourceId.REGEX + "regions/([^/]+)";
+  private static final Pattern PATTERN = Pattern.compile(REGEX);
+  private static final long serialVersionUID = 5569092266957249294L;
+
+  private final String region;
+
+  private RegionId(String project, String region) {
+    super(project);
+    this.region = checkNotNull(region);
+  }
+
+  private RegionId(RegionId regionId) {
+    super(regionId.project());
+    this.region = checkNotNull(regionId.region());
+  }
+
+  /**
+   * Returns the name of the region.
+   */
+  public final String region() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary final modifier in final class](https://www.codacy.com/app/mziccard/gcloud-java/file/2745817047/issues/source?bid=3123107&fileBranchId=3356751#l65)